src/parser/jobs/smn/modules/Swiftcast.js

```diff
@@ -8,5 +10,48 @@ const MISSED_SEVERITIES = {
 }
 
 export default class Swiftcast extends SwiftcastModule {
+	static handle = 'swiftcast'
+
 	severityTiers = MISSED_SEVERITIES
+	_dwtActive = false
+	_dwtActiveTimestamp = 0
+
+	constructor(...args) {
+		super(...args)
+
+		this.addEventHook('cast', {to: 'player'}, this._onCast)
+		this.addEventHook('normaliseddamage', {by: 'player', abilityId: this.data.actions.DEATHFLARE.id}, this._endDWT)
+		this.addEventHook('death', {to: 'player'}, this._endDWT)
+	}
+
+	_onCast(event) {
+		const guid = event.ability.guid
+
+		if (guid === this.data.actions.DREADWYRM_TRANCE.id) {
+			this.debug(`dwt now active ${this.parser.formatTimestamp(event.timestamp)}`)
+			this._dwtActive = true
+			this._dwtActiveTimestamp = event.timestamp
+			return
+		}
+
+		if (this._dwtActive && event.timestamp - this._dwtActiveTimestamp > DWT_LENGTH) {
+			this._endDWT(event)
+		}
+	}
+
+	_endDWT(event) {
+		this.debug(`Ending DWT ${this.parser.formatTimestamp(event.timestamp)}`)
+		this._dwtActive = false
+	}
+
+	considerSwiftAction(event) {
+		this.debug(`DWT Active (${this._dwtActive}): ${event.name}`)
+		if (this._dwtActive) {
+			if (event.castTime + DWT_CAST_TIME_MOD > 0) {
+				return true
+			}
+			return false
+		}
+		return true
```

Comment:
Could this just not be one conditional, kinda iffy to read, something like this:

```
if (this._dwtActive && event.castTime + DWT_CAST_TIME_MOD <= 0) {
  return false
}

return true
```

Reply:
That was the one that was hanging me up at midnight-thirty (if you read the caster channel I think my brain BSoD'd at that time) – will change.